drivers/platform/aducm3029/irq.c

```diff
@@ -181,14 +196,28 @@ int32_t irq_register_callback(struct irq_ctrl_desc *desc, uint32_t irq_id,
  */
 int32_t irq_unregister(struct irq_ctrl_desc *desc, uint32_t irq_id)
 {
+	struct aducm_irq_ctrl_desc	*aducm_desc;
+	struct uart_desc		*uart_desc;
+
 	if (!desc || !desc->extra || !initialized ||
-	    irq_id >= NB_EXT_INTERRUPTS)
+	    irq_id >= NB_INTERRUPTS)
 		return FAILURE;
 
-	((struct aducm_irq_ctrl_desc *)desc->extra)->
-			callbacks[irq_id].callback = 0;
+	aducm_desc = desc->extra;
 
-	return SUCCESS;
+	if (irq_id < NB_EXT_INTERRUPTS) {
+		adi_xint_RegisterCallback(id_map_event[irq_id], 0, 0);
+	} else if (aducm_desc->conf[irq_id].uart_conf) { //irq_id = UART_ID
```

Comment:
nitpick: we probably want to use C style comments to keep the consistency.

Reply:
A switch will be used here too